### PR TITLE
feat: add `ExecuteWithin`

### LINKED ISSUE #50: Add expectations for execution time
For delegate types a natural enhancement would be to test that the delegate executed within a certain amount of time, e.g.
```csharp
void Action() => DoSomething();
await Expect.That(Action).Should().ExecuteWithin(200.Milliseconds());
```

### PR DESCRIPTION
**Pull request by vbreuss**
Fixes #50 by adding an `ExecuteWithin(TimeSpan)` method for delegates. The delegates are executed within a `Stopwatch` and the time measured, that it takes to completion.

Additionally add delegate overloads that take a `CancellationToken` as parameter, that can be set via the `WithCancellation` functionality.

In order to enable testing of time-related functionality, the Stopwatch is wrapped in an interface and an internal method allows overwriting this. Internals are visible to a specific project for testing edge cases: `Testably.Expectations.Internal.Tests`

### CODE REVIEW

File: Source/Testably.Expectations/Results/ExpectationResult.cs [hunk 1]
```diff
@@ -47,6 +48,12 @@ public ExpectationResult WithCancellation(CancellationToken cancellationToken,
 		return this;
 	}
 
+	internal ExpectationResult UseTimeSystem(ITimeSystem timeSystem)
```
Comment (vbreuss) [inline Source/Testably.Expectations/Results/ExpectationResult.cs:51]: Add XML-Comments

File: Source/Testably.Expectations/That/Delegates/ThatDelegateShould.ExecuteWithin.cs [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+					$"it took only {Formatter.Format(actual.Duration)}");
+			}
+
+			throw new InvalidOperationException("Received <null> in ExecuteWithinConstraint.");
+		}
+
+		public override string ToString()
+			=> $"not execute within {Formatter.Format(duration)}";
+	}
+	private readonly struct NotExecuteWithinConstraint(TimeSpan duration) : IValueConstraint<DelegateValue>
```
Comment (vbreuss) [inline Source/Testably.Expectations/That/Delegates/ThatDelegateShould.ExecuteWithin.cs:138]: Can these 4 classes be simplified/combined?

File: Source/Testably.Expectations/That/Delegates/ThatDelegateShould.ExecuteWithin.cs [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+		[CallerArgumentExpression("duration")] string doNotPopulateThisValue = "")
+		=> new(source.ExpectationBuilder
+			.AddConstraint(new NotExecuteWithinConstraint(duration))
+			.AppendMethodStatement(nameof(NotExecuteWithin), doNotPopulateThisValue));
+
+	private readonly struct ExecuteWithinConstraint<TValue>(TimeSpan duration) : IValueConstraint<DelegateValue<TValue>>
+	{
+		public ConstraintResult IsMetBy(DelegateValue<TValue>? actual)
+		{
+			if (actual?.Exception is { } exception)
```
Comment (vbreuss) [inline Source/Testably.Expectations/That/Delegates/ThatDelegateShould.ExecuteWithin.cs:63]: Add tests for how exceptions are handled
